src/FileWriter.php

```diff
@@ -25,21 +26,25 @@ final class FileWriter
      */
     public static function writeFile($file, $content, $chmod = 0666)
     {
-        $tmp = tempnam(sys_get_temp_dir(), md5($file));
+        $dir = dirname($file);
+        $tmp = tempnam($dir, md5($file));
 
         if (file_put_contents($tmp, $content) === false) {
             throw Exception\WriteContentException::unableToWriteContent($tmp);
         }
 
         if (chmod($tmp, $chmod & ~umask()) === false) {
+            @unlink($tmp);
```

Comment:
Is it necessary to suppress this error? If something goes wrong here, I'd rather have all possible information logged than suppressed.